plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/Processor.java

```diff
@@ -28,22 +28,38 @@
  * A processor implementation may modify the data belonging to a document.
  * Whether changes are made and what exactly is modified is up to the implementation.
  */
-public interface Processor {
+public abstract class Processor {
```

Comment:
maybe `Processor` should remain to be an interface? We just add `getId()` to the interface. Each implementation would then need to have a `id` field.